docs/utils.md

```diff
@@ -23,3 +23,55 @@ $foo = $client->get('http://example.com/foo');
 $bar = $client->get('http://example.com/bar', ['accept-encoding' => 'application/json']);
 $post = $client->post('http://example.com/update', [], 'My post body');
 ```
+
+## BatchClient
+
+This client wraps a HttpClient and extends it with the possibility to send an array of requests and to retrieve their responses as a `BatchResult`.
+
+``` php
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\MessageFactoryDiscovery;
+
+$messageFactory = MessageFactoryDiscovery::find();
+
+$requests = [
+    $messageFactory->createRequest('GET', 'http://example.com/foo'),
+    $messageFactory->createRequest('POST', 'http://example.com/update', [], 'My post body'),
+];
+
+$client = new BatchClient(
+    HttpClientDiscovery::find()
+);
+
+$batchResult = $client->sendRequests($requests);
+```
+
+The `BatchResult` itself is an object that contains responses for all requests sent. It provides methods that give appropriate information based on a given request.
+
+``` php
+$requests = [
+    $messageFactory->createRequest('GET', 'http://example.com/foo'),
+    $messageFactory->createRequest('POST', 'http://example.com/update', [], 'My post body'),
+];
+
+$batchResult = $client->sendRequests($requests);
+
+if ($batchResult->hasResponses()) {
+    $fooSuccessful = $batchResult->isSuccesful($requests[0]);
+    $updateResponse = $batchResult->getResponseFor($request[1]);
```

Comment:
this is assuming that the post request did not fail and would throw an exception if it failed. in this flow of code, that can't happen as we would have bailed out with an exception. but that means $fooSuccessful is pointless. i would move the isSuccessful check below to the error handling logic, as its only relevant in the error case to figure out if something went wrong or not.